src/cascadia/TerminalSettingsModel/CascadiaSettings.h

```diff
@@ -199,6 +208,74 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         mutable std::once_flag _commandLinesCacheOnce;
         mutable std::vector<std::pair<std::wstring, Model::Profile>> _commandLinesCache;
     };
+
+    struct FragmentProfileEntry : FragmentProfileEntryT<FragmentProfileEntry>
+    {
+    public:
+        FragmentProfileEntry(winrt::guid profileGuid, hstring json) :
+            _profileGuid{ profileGuid },
+            _json{ json } {}
+
+        winrt::guid ProfileGuid() const noexcept { return _profileGuid; }
+        hstring Json() const noexcept { return _json; }
+
+    private:
+        winrt::guid _profileGuid;
+        hstring _json;
+    };
+
+    struct FragmentColorSchemeEntry : FragmentColorSchemeEntryT<FragmentColorSchemeEntry>
+    {
+    public:
+        FragmentColorSchemeEntry(hstring schemeName, hstring json) :
+            _schemeName{ schemeName },
+            _json{ json } {}
+
+        hstring ColorSchemeName() const noexcept { return _schemeName; }
+        hstring Json() const noexcept { return _json; }
+        WINRT_PROPERTY(bool, Conflict, false);
+
+    private:
+        hstring _schemeName;
+        hstring _json;
+    };
+
+    struct FragmentSettings : FragmentSettingsT<FragmentSettings>
+    {
+    public:
+        FragmentSettings(hstring source, hstring json, hstring jsonSource, FragmentScope scope) :
+            _source{ source },
+            _json{ json },
+            _jsonSource{ jsonSource },
+            _scope{ scope },
+            _modifiedProfiles{ winrt::single_threaded_vector<Model::FragmentProfileEntry>() },
+            _newProfiles{ winrt::single_threaded_vector<Model::FragmentProfileEntry>() },
+            _colorSchemes{ winrt::single_threaded_vector<Model::FragmentColorSchemeEntry>() } {}

```

Comment:
we will allocate three blank vectors for every kind of fragment and dynamic profile generator, the majority of which will never populate more than one of them.